Mathlib/CategoryTheory/Generator.lean

```diff
@@ -351,6 +363,14 @@ def IsDetector (G : C) : Prop :=
 def IsCodetector (G : C) : Prop :=
   IsCodetecting ({G} : Set C)
 
+
+section Equivalence
+
+theorem IsSeparator.of_equivalence {G : C} (h : IsSeparator G) (F : C ⥤ D) [F.IsEquivalence] :
+    IsSeparator (F.obj G) := by simpa using IsSeparating.of_equivalence h (F.asEquivalence)
+
+end Equivalence
+
```

Comment:
As there are no new variables and there is only one declaration, we may remove `section`. Also, by uniformity with `HasSeparator.of_equivalence`, it is probably better to use an equivalence of categories rather than a functor which is an equivalence:
```suggestion
theorem IsSeparator.of_equivalence {G : C} (h : IsSeparator G) (α : C ≌ D) :
    IsSeparator (α.functor.obj G) := by simpa using IsSeparating.of_equivalence h α
```

Reply:
In the meantime and before reading your suggestion, I added the dualized version `IsCoseparator.of_equivalence`, so now there are two declarations. Is it fine now to keep the section?